### PR TITLE
Chapter 1

### PR DESCRIPTION
**Pull request by Sibirtsev**
### Solutions for Chapter 1

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -608,14 +614,19 @@ Implement a function that returns the sum of the last two digits of a number.
...
 specifying complex expressions.
 -}
 
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n = 
+  fstDig + sndDig
+  where
+    fstDig = mod (abs n) 10
+    sndDig = mod (div ((abs n) - fstDig) 10) 10
```
Comment (chshersh) [inline src/Chapter1.hs]: On a side note, there is a standard function as well that combines both `div` and `mod` at once: `divMod`.

It returns the pair, so you can pattern match on that and get both variabless at once:

```haskell
let (a, b) = abs n `divMod` 10
```



That is a wonderful solution! 👏🏼  You correctly noticed that it is the `div` and `mod`, cool 😎 

One hint to make your solution even shorter: the standard library has the [`divMod`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Prelude.html#v:divMod) function, that is actually combines inside both `div` and `mod`. And this is exactly what you use!.

So you could write it this way:

```haskell
(first, second) = divMod (mod x 100) 10
```

You can see how we could pattern match on the pair 🙂

File: src/Chapter1.hs [hunk 2]
```diff
@@ -635,8 +646,10 @@ Implement a function that returns the first digit of a given number.
 You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
-
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n 
+    | (abs n) < 10 = abs n
+    | otherwise = firstDigit (div (abs n) 10)
```
Comment (chshersh) [inline src/Chapter1.hs]: Nice solution! I see that you use `abs n` several times. You can move it into a separate variable, to reduce some `()` and more code a bit cleaner :slightly_smiling_face:

**Review by chshersh — APPROVED** [review]
Excellent solutions! I enjoyed reviewing them a lot :hugs: 
And very well done on the Advanced Task :medal_sports: 

I left a few comments, but they are just minor notes and some thoughts. I think this PR is ready for merge  :checkered_flag:

**Comment by chshersh** [comment]
⚠️ By the way, per new rules of the Hacktoberfest event, you would need to add the `hacktoberfest` topic to this fork repo:

* https://hacktoberfest.digitalocean.com/hacktoberfest-update

All previous contributions are legit, but from now on it should have the label to participate.

**Comment by Sibirtsev** [comment]
@chshersh I refactored both PRs  
Thank you a lot for your comments and for the challenge itself. Will continue with the next chapter :)

**Pull request merged**